packages/xrpl/test/models/CredentialCreate.test.ts

```diff
@@ -0,0 +1,232 @@
+      errorMessage,
+    )
+    assert.throws(
+      () => validate(credentialCreate),
+      ValidationError,
+      errorMessage,
+    )
+  })
+
+  it(`throws w/ credentialType field too long`, function () {
+    credentialCreate.CredentialType = stringToHex(
+      'PassportPassportPassportPassportPassportPassportPassportPassportPassportPassportPassportPassportPassportPassportPassportPassportPassportPassport',
+    )
+    const errorMessage = 'CredentialCreate: CredentialType length must be < 128'
+    assert.throws(
+      () => validateCredentialCreate(credentialCreate),
+      ValidationError,
+      errorMessage,
+    )
+    assert.throws(
+      () => validate(credentialCreate),
+      ValidationError,
+      errorMessage,
+    )
+  })
+
+  it(`throws w/ credentialType field empty`, function () {
+    credentialCreate.CredentialType = ''
+    const errorMessage = 'CredentialCreate: CredentialType length must be > 0'
+    assert.throws(
+      () => validateCredentialCreate(credentialCreate),
+      ValidationError,
+      errorMessage,
+    )
+    assert.throws(
+      () => validate(credentialCreate),
+      ValidationError,
+      errorMessage,
+    )
+  })
+
+  it(`throws w/ credentialType field not hex`, function () {
+    credentialCreate.CredentialType = 'this is not hex'
+    const errorMessage =
+      'CredentialCreate: CredentialType must be encoded in hex'
+    assert.throws(
+      () => validateCredentialCreate(credentialCreate),
+      ValidationError,
+      errorMessage,
+    )
+    assert.throws(
+      () => validate(credentialCreate),
+      ValidationError,
+      errorMessage,
+    )
+  })
+
+  it(`throws w/ Expiration field not number`, function () {
+    credentialCreate.Expiration = 'this is not a number'
+    const errorMessage = 'CredentialCreate: invalid field Expiration'
+    assert.throws(
+      () => validateCredentialCreate(credentialCreate),
+      ValidationError,
+      errorMessage,
+    )
+    assert.throws(
+      () => validate(credentialCreate),
+      ValidationError,
+      errorMessage,
+    )
+  })
+
+  it(`throws w/ URI field not a string`, function () {
+    credentialCreate.URI = 123
+    const errorMessage = 'CredentialCreate: invalid field URI'
+    assert.throws(
+      () => validateCredentialCreate(credentialCreate),
+      ValidationError,
+      errorMessage,
+    )
+    assert.throws(
+      () => validate(credentialCreate),
+      ValidationError,
+      errorMessage,
+    )
+  })
+
+  it(`throws w/ URI field empty`, function () {
+    credentialCreate.URI = ''
+    const errorMessage = 'CredentialCreate: URI length must be > 0'
+    assert.throws(
+      () => validateCredentialCreate(credentialCreate),
+      ValidationError,
+      errorMessage,
+    )
+    assert.throws(
+      () => validate(credentialCreate),
+      ValidationError,
+      errorMessage,
+    )
+  })
+
+  it(`throws w/ URI field too long`, function () {
+    credentialCreate.URI = stringToHex(
```

Comment:
`credentialCreate.URI = stringToHex('A' * 33)` -- this is slightly easier to read.
I'm assuming that the URI field has an upper bound of 64 bytes in its representation.